src/components/PostDetails.tsx

```diff
@@ -1,92 +1,60 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { Loader } from './Loader';
+import { useAppSelector, useAppDispatch } from '../app/hooks';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import {
+  fetchComments,
+  removeComment,
+  addComment,
+} from '../features/comments/commentsSlice';
+import { useSelector } from 'react-redux';
 
 type Props = {
   post: Post;
 };
 
+interface Post {
+  id: number;
+  title: string;
+  body: string;
+}
+
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const {
+    loaded,
+    hasError,
+    items: comments,
+  } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
+  const previousPostId = useRef(post.id);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  useEffect(() => {
+    dispatch(fetchComments(post.id));
+  }, [dispatch, post.id]);
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+  useEffect(() => {
     setVisible(false);
-    setError(false);
+  }, [post.id]);
 
+  const handleAddComment = async ({ name, email, body }: CommentData) => {
     try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
+      await dispatch(
+        addComment({ name, email, body, postId: post.id }),
+      ).unwrap();
+    } catch (er) {
```

Comment:
The catch block is empty. Consider adding error handling logic to provide feedback to the user when an error occurs during comment addition.